lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/BuildCriOauthRequestHandler.java

```diff
@@ -157,14 +158,15 @@ public Map<String, Object> handleRequest(JourneyRequest input, Context context)
                                 ErrorResponse.MISSING_CREDENTIAL_ISSUER_ID)
                         .toObjectMap();
             }
-            LogHelper.attachCriIdToLogs(criId);
+            var cri = Cri.fromId(criId);
+            LogHelper.attachCriIdToLogs(cri);
 
             String criContext = getJourneyParameter(input, CONTEXT);
             EvidenceRequest criEvidenceRequest =
                     EvidenceRequest.fromBase64(getJourneyParameter(input, EVIDENCE_REQUESTED));
-            String connection = configService.getActiveConnection(criId);
+            String connection = configService.getActiveConnection(cri);
             OauthCriConfig criConfig =
-                    configService.getOauthCriConfigForConnection(connection, criId);
+                    configService.getOauthCriConfigForConnection(connection, cri);
 
             if (criConfig == null) {
```

Comment:
You might need to pull this up (or replace it) - since it's effectively being superseded by`Cri.fromId`.

I wonder if it's worth pushing inside `getCriFromJourney`, and that can do all the relevant null checking/validity checking in one place.

Reply:
I did initially have it in there but wasn't able to distinguish between a `MISSING_CREDENTIAL_ISSUER_ID` and `INVALID_CREDENTIAL_ISSUER_ID` errors - can we just use `INVALID_CREDENTIAL_ISSUER_ID` for both?

Reply:
Sounds fine to me